tests/main/listing/task.yaml

```diff
@@ -14,15 +14,15 @@ restore: |
 
 execute: |
     echo "List prints core snap version"
-    # most core versions should be like "16-2", so [0-9]{2}-[0-9.]+
-    # but edge will have a timestamp in there, "16.2+201701010932", so add an optional \+[0-9]+ to the end
-    # *current* edge also has .git. and a hash snippet, so add an optional .git.[0-9a-f]+ to the already optional timestamp
+    # all core versions will start with something like "16-2.61.4", so "[0-9]{2}-[0-9.]+"
+    # with core in security maintenance it has additional date like "16-2.61.4-20240607", so optional "(-[0-9]+)?"
+    # in edge it has additional git identifier like "16-2.63+git5348.e45449bd5", so optional "(\\+git[0-9]+\\.[0-9a-f]+)?"
 
     # Expressions for version and revision
```

Comment:
Please double check:
- Assume "16.2~xxxxx" is not relevant anymore for both core cases and snapd, removed "~[a-z0-9]+)?"
- Assume "16.2+201701010932.xxxxx" for core (prev FINAL_VERSION) is not relevant anymore or was wrong, removed "(\\+[0-9]+\\.[0-9a-f]+)?"

Main change:
- Added "(-[0-9]+)?" for core to support "16-2.61.4-20240607" used for security maintenance